pkg/operator/ceph/object/user/controller.go

```diff
@@ -18,202 +18,308 @@ limitations under the License.
+		cephObjectStoreUser.Status.Phase = k8sutil.ReconcileFailedStatus
+		errStatus := opcontroller.UpdateStatus(r.client, cephObjectStoreUser)
+		if errStatus != nil {
+			return reconcile.Result{}, errors.Wrap(errStatus, "failed to set status")
+		}
+		return reconcileResponse, err
 	}
 
-	if err = deleteUser(c.context, user); err != nil {
-		logger.Errorf("failed to delete object store user %q. %v", user.Name, err)
+	// CREATE/UPDATE KUBERNETES SECRET
+	reconcileResponse, err = r.reconcileCephUserSecret(cephObjectStoreUser)
+	if err != nil {
+		cephObjectStoreUser.Status.Phase = k8sutil.ReconcileFailedStatus
+		errStatus := opcontroller.UpdateStatus(r.client, cephObjectStoreUser)
+		if errStatus != nil {
+			return reconcile.Result{}, errors.Wrap(errStatus, "failed to set status")
+		}
+		return reconcileResponse, err
 	}
-}
 
-// ParentClusterChanged determines wether or not a CR update has been sent
-func (c *ObjectStoreUserController) ParentClusterChanged(cluster cephv1.ClusterSpec, clusterInfo *cephconfig.ClusterInfo, isUpgrade bool) {
-	logger.Debugf("No need to update object store users after the parent cluster changed")
+	// Set Ready status, we are done reconciling
+	cephObjectStoreUser.Status.Phase = k8sutil.ReadyStatus
+	err = opcontroller.UpdateStatus(r.client, cephObjectStoreUser)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to set status")
+	}
+
+	// Return and do not requeue
+	logger.Debug("done reconciling")
+	return reconcile.Result{}, nil
 }
 
-func (c *ObjectStoreUserController) storeUserOwners(store *cephv1.CephObjectStoreUser) []metav1.OwnerReference {
-	// Only set the cluster crd as the owner of the object store user resources.
-	// If the object store user crd is deleted, the operator will explicitly remove the object store user resources.
-	// If the object store user crd still exists when the cluster crd is deleted, this will make sure the object store user
-	// resources are cleaned up.
-	return []metav1.OwnerReference{c.ownerRef}
+func (r *ReconcileObjectStoreUser) reconcileCephUser(cephObjectStoreUser *cephv1.CephObjectStoreUser) (reconcile.Result, error) {
+	err := r.createCephUser(cephObjectStoreUser)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to create object store user %q", cephObjectStoreUser.Name)
+	}
+
+	return reconcile.Result{}, nil
 }
 
-func getObjectStoreUserObject(obj interface{}) (objectstoreuser *cephv1.CephObjectStoreUser, err error) {
-	var ok bool
-	objectstoreuser, ok = obj.(*cephv1.CephObjectStoreUser)
-	if ok {
-		// the objectstoreuser object is of the latest type, simply return it
-		return objectstoreuser.DeepCopy(), nil
+func (r *ReconcileObjectStoreUser) createCephUser(u *cephv1.CephObjectStoreUser) error {
+	logger.Infof("creating ceph object user %q in namespace %q", u.Name, u.Namespace)
+	user, rgwerr, err := object.CreateUser(r.objContext, r.userConfig)
+	if err != nil {
+		if strings.Contains(err.Error(), "user already exists") {
```

Comment:
Is there anything more reliable such as an error code instead of a substring from the error message? Error messages tend to be brittle.

Reply:
No really, other portions of the code do this too.

Reply:
Yeah, was just hoping to reduce these, rather than see them grow. `rgwerr` isn't meaningful if it already exists?

Reply:
@leseb What is `rgwerr` when the user already exists?

Reply:
`17`

Reply:
This seems to be quite reliable since it follows standard convention:
`#define EEXIST          17      /* File exists */
`.

Reply:
But we don't have the return code unfortunately...